Add draft feature to private messages.

zachleigh opened the pull request:
This pull request addresses issue #32.

trang commented:
It's looking good :) Just a few things to fix:

1. You want to make sure that users cannot see other people's drafts. At the moment, if I save a message as draft, copy the URL, log in as another user, and go the draft URL I copied, I can see the message.

2. When viewing a draft message, the label above the form is always "Reply", even when I save a draft for a new message. You may want to make the label more general, like "Message".

3. It would be better to remove the `md-primary` for the "Save as draft" button, so that it looks less important than the "Send" button. in general, we don't want several primary buttons, only one should be primary.

zachleigh commented:
Fixed the issues and changed one other thing.     

1. Enforced draft privacy by checking to make sure the private message's sender id is equal to the current user id in the controller `write` method.

2. Set the label to 'Message'.

3. Removed the `md-primary` class. The button is now white, which looks fine to me and moves focus to the Send button as desired.

4. When saving a draft, the To field of the message is no longer required. Only the content field needs to be filled.
